src/include/cursor_inline.h

```diff
@@ -352,10 +352,14 @@ __wt_cursor_dhandle_decr_use(WT_SESSION_IMPL *session)
 
     dhandle = session->dhandle;
 
-    /* If we close a handle with a time of death set, clear it. */
+    /*
+     * If we close a handle with a time of death set, clear it. The ordering is important: after
+     * decrementing the use count, there's a chance that the data handle can be freed.
+     */
     WT_ASSERT(session, dhandle->session_inuse > 0);
-    if (__wt_atomic_subi32(&dhandle->session_inuse, 1) == 0 && dhandle->timeofdeath != 0)
+    if (dhandle->timeofdeath != 0 && dhandle->session_inuse == 1)
         dhandle->timeofdeath = 0;
+    (void)__wt_atomic_subi32(&dhandle->session_inuse, 1);
```

Comment:
Can you confirm we don't get into trouble if we set `timeofdeath` to 0, but then `session_inuse` isn't decremented to 0, that is, another thread increments `session_inuse` between our setting `timeofdeath` and our decrement of `session_inuse`?

I took a look and don't see any problems, but wanted to confirm with you.

Reply:
I don't think this is a problem.  Even if it's not perfect with an occasional race (so it's set or not set inappropriately) it doesn't much matter.  Setting it to 0 communicates to the sweep thread - hey someone just used this handle, don't mark it dead yet.  So it has heuristic value only.